lib/path.cpp

```diff
@@ -358,16 +358,25 @@ bool Path::isHeader(const std::string &path)
 
 std::string Path::getAbsoluteFilePath(const std::string& filePath)
 {
+    if (filePath.empty())
+        return "";
+
     std::string absolute_path;
 #ifdef _WIN32
     char absolute[_MAX_PATH];
     if (_fullpath(absolute, filePath.c_str(), _MAX_PATH))
         absolute_path = absolute;
+    if (absolute_path.back() == '\\')
```

Comment:
there is a good chance that absolute_path is empty right? don't call back() in that case.

Reply:
That would indeed be undefined: https://en.cppreference.com/w/cpp/string/basic_string/back - and we even report that. I wonder if that might exists in other parts of the code. We could probably be more aggressive about detecting that - maybe similar to a NULL dereference.

The `_fullpath()` call might need better errorhandling.

Reply:
> That would indeed be undefined: https://en.cppreference.com/w/cpp/string/basic_string/back - and we even report that. I wonder if that might exists in other parts of the code. We could probably be more aggressive about detecting that - maybe similar to a NULL dereference.

This is not reported because it is in the `_WIN32` code which we do not analyze in the CI. We should probably add  analysis steps for files using those to the selfchecks.

Reply:
Fixed. I also added a bunch of missing test cases and TODOs.

I also filed https://trac.cppcheck.net/ticket/13158 about enhancing the selfcheck.